azure-ts-dynamicresource/cdnCustomDomain.ts

```diff
@@ -0,0 +1,276 @@
+
+    async check(olds: DynamicProviderInputs, news: DynamicProviderInputs): Promise<pulumi.dynamic.CheckResult> {
+        const cdnClient = await this.getCDNManagementClient();
+        let result: pulumi.dynamic.CheckResult;
+
+        // If none of the CDN properties changed, then there is nothing to be validated.
+        if (olds.profileName === news.profileName &&
+            olds.endpointName === news.endpointName &&
+            olds.customDomainHostName === news.customDomainHostName) {
+            return {
+                inputs: news
+            };
+        }
+
+        const validationOutput = await cdnClient.endpoints.validateCustomDomain(
+            news.resourceGroupName, news.profileName, news.endpointName, news.customDomainHostName);
+        if (!validationOutput.customDomainValidated) {
+            result = {
+                failures: [
+                    {
+                        property: this.name,
+                        reason: validationOutput.reason || "domain_validation_failed"
+                    }
+                ]
+            };
+        } else {
+            result = {
+                inputs: news
+            };
+        }
+
+        return result;
+    }
+
+    async diff(id: string, previousOutput: DynamicProviderOutputs, news: DynamicProviderInputs): Promise<pulumi.dynamic.DiffResult> {
+        const replaces: string[] = [];
+        let changes = false;
+        let deleteBeforeReplace = false;
+
+        if (previousOutput.customDomainHostName !== news.customDomainHostName || previousOutput.name !== this.name) {
+            console.warn("Changing the domain name properties will cause a downtime.")
+
+            deleteBeforeReplace = true;
+            if (previousOutput.customDomainHostName !== news.customDomainHostName) {
+                replaces.push("customDomainHostName");
+            }
+            if (previousOutput.name !== this.name) {
+                replaces.push("customDomainName");
+            }
+        }
+
+        if (previousOutput.endpointName !== news.endpointName) {
+            deleteBeforeReplace = true;
+            replaces.push("endpointName");
+        }
+
+        // HTTPS can be enabled/disabled in-place.
+        if (previousOutput.httpsEnabled !== news.httpsEnabled) {
+            changes = true;
+            replaces.push("httpsEnabled");
+        }
+
+        return {
+            deleteBeforeReplace: deleteBeforeReplace,
+            replaces: replaces,
+            changes: changes
+        };
+    }
+
+    async create(inputs: DynamicProviderInputs): Promise<pulumi.dynamic.CreateResult> {
+        const cdnClient = await this.getCDNManagementClient();
+        const result =
+            await cdnClient.customDomains
+                .create(
+                    inputs.resourceGroupName,
+                    inputs.profileName,
+                    inputs.endpointName,
+                    this.name,
+                    inputs.customDomainHostName);
```

Comment:
heavy wrapping.  consider just:

```ts
const result = await cdnClient.customDomains.create(
    inputs.resourceGroupName,
    inputs.profileName,
    inputs.endpointName,
    this.name,
    inputs.customDomainHostName);
```

Or, if it's not too long:

```ts
const result = await cdnClient.customDomains.create(
    inputs.resourceGroupName, inputs.profileName, inputs.endpointName, this.name, inputs.customDomainHostName);
```

Reply:
Also, in general, in pulumi resources and whatnot, we have 'name' come first.

Reply:
Sure I did this here and in a couple of other places.